src/pybind/mgr/dashboard/frontend/src/app/ceph/block/mirroring/bootstrap-import-modal/bootstrap-import-modal.component.ts

```diff
@@ -67,32 +65,41 @@ export class BootstrapImportModalComponent extends BaseModal implements OnInit,
 
     this.subs = this.rbdMirroringService.subscribeSummary((data) => {
       const pools = data.content_data.pools;
+      console.log("pools",pools)
       this.pools = pools.reduce((acc: any[], pool: Pool) => {
         acc.push({
           name: pool['name'],
-          mirror_mode: pool['mirror_mode']
+          mirror_mode: pool['mirror_mode'],
+          peer_uuids : pool['peer_uuids']
         });
         return acc;
       }, []);
 
       const poolsControl = this.importBootstrapForm.get('pools') as UntypedFormGroup;
       _.each(this.pools, (pool) => {
+        console.log(pool)
         const poolName = pool['name'];
         const mirroring_disabled = pool['mirror_mode'] === 'disabled';
         const control = poolsControl.controls[poolName];
-        if (control) {
-          if (mirroring_disabled && control.disabled) {
-            control.enable();
-          } else if (!mirroring_disabled && control.enabled) {
-            control.disable();
-            control.setValue(true);
+          if (mirroring_disabled || pool.peer_uuids.length > 0) {
+            if (control) {
+                control.disable()
+            } else {
+              poolsControl.addControl(
+                poolName,
+                new UntypedFormControl({ value: false, disabled: true })
+              );
+            }
+          } else {
+            if (control) {
+              control.enable();
+            } else {
+              poolsControl.addControl(
+                poolName,
+                new UntypedFormControl({ value: false, disabled: false })
+              );
+            }
```

Comment:
This logic seems repeated, can we create a helper function for this and call at both places ?

Reply:
its not repeated, I have set different set of values according to the scenario